[Snap] Snap builds failing in automation

STR:
- Push a commit to master

Expected:
- The snapcraft remote build servers are automatically triggered to generate builds for all architectures
- Snaps for all architectures are successfully built and released on the edge channel in the Snap Store

Actual:
- Builds fail in automation with the following error:

```
Building webthings-gateway
:: + craftctl default
:: ++ npm --version
:: /build/webthings-gateway/parts/webthings-gateway/run/build.sh: line 5: npm: command not found
:: + NPM_VERSION=
:: error:
:: Traceback (most recent call last):
::   File "/snap/snapcraft/current/lib/python3.10/site-packages/craft_parts/ctl.py", line 111, in main
::     ret = CraftCtl.run(cmd, args)
::   File "/snap/snapcraft/current/lib/python3.10/site-packages/craft_parts/ctl.py", line 45, in run
::     _client(cmd, args)
::   File "/snap/snapcraft/current/lib/python3.10/site-packages/craft_parts/ctl.py", line 98, in _client
::     raise RuntimeError(message)
:: RuntimeError: Failed to run the build script for part 'webthings-gateway'.
'override-build' in part 'webthings-gateway' failed with code 1.
```

Note: The snap was successfully building locally for the AMD64 architecture using the `snapcraft` command.

@ogra1 Thanks very much for that patch.

- Good news: Now building successfully in automation for AMD64
- Bad news: Builds are still failing on all other architectures, including ARM64:

```
:: > segfault-handler@1.3.0 install /build/webthings-gateway/parts/webthings-gateway/build/node_modules/segfault-handler
:: > node-gyp rebuild
::
:: make: Entering directory '/build/webthings-gateway/parts/webthings-gateway/build/node_modules/segfault-handler/build'
::   CXX(target) Release/obj.target/segfault-handler/src/segfault-handler.o
:: In file included from ../../nan/nan.h:60,
::                  from ../src/segfault-handler.cpp:2:
:: /root/.cache/node-gyp/10.24.1/include/node/node.h:573:7: warning: cast between incompatible function types from ‘void (*)(Nan::ADDON_REGISTER_FUNCTION_ARGS_TYPE)’ {aka ‘void (*)(v8::Local<v8::Object>)’} to ‘node::addon_register_func’ {aka ‘void (*)(v8::Local<v8::Object>, v8::Local<v8::Value>, void*)’} [-Wcast-function-type]
::   573 |       (node::addon_register_func) (regfunc),                          \
::       |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
:: /root/.cache/node-gyp/10.24.1/include/node/node.h:607:3: note: in expansion of macro ‘NODE_MODULE_X’
::   607 |   NODE_MODULE_X(modname, regfunc, NULL, 0)  // NOLINT (readability/null_usage)
::       |   ^~~~~~~~~~~~~
:: ../src/segfault-handler.cpp:346:3: note: in expansion of macro ‘NODE_MODULE’
::   346 |   NODE_MODULE(segfault_handler, init)
::       |   ^~~~~~~~~~~
:: In file included from /root/.cache/node-gyp/10.24.1/include/node/node.h:63,
::                  from ../../nan/nan.h:60,
::                  from ../src/segfault-handler.cpp:2:
:: /root/.cache/node-gyp/10.24.1/include/node/v8.h: In instantiation of ‘void v8::PersistentBase<T>::SetWeak(P*, typename v8::WeakCallbackInfo<P>::Callback, v8::WeakCallbackType) [with P = node::ObjectWrap; T = v8::Object; typename v8::WeakCallbackInfo<P>::Callback = void (*)(const v8::WeakCallbackInfo<node::ObjectWrap>&)]’:
:: /root/.cache/node-gyp/10.24.1/include/node/node_object_wrap.h:84:25:   required from here
:: /root/.cache/node-gyp/10.24.1/include/node/v8.h:9502:16: warning: cast between incompatible function types from ‘v8::WeakCallbackInfo<node::ObjectWrap>::Callback’ {aka ‘void (*)(const v8::WeakCallbackInfo<node::ObjectWrap>&)’} to ‘Callback’ {aka ‘void (*)(const v8::WeakCallbackInfo<void>&)’} [-Wcast-function-type]
::  9502 |                reinterpret_cast<Callback>(callback), type);
::       |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
:: /root/.cache/node-gyp/10.24.1/include/node/v8.h: In instantiation of ‘void v8::PersistentBase<T>::SetWeak(P*, typename v8::WeakCallbackInfo<P>::Callback, v8::WeakCallbackType) [with P = Nan::ObjectWrap; T = v8::Object; typename v8::WeakCallbackInfo<P>::Callback = void (*)(const v8::WeakCallbackInfo<Nan::ObjectWrap>&)]’:
:: ../../nan/nan_object_wrap.h:64:57:   required from here
:: /root/.cache/node-gyp/10.24.1/include/node/v8.h:9502:16: warning: cast between incompatible function types from ‘v8::WeakCallbackInfo<Nan::ObjectWrap>::Callback’ {aka ‘void (*)(const v8::WeakCallbackInfo<Nan::ObjectWrap>&)’} to ‘Callback’ {aka ‘void (*)(const v8::WeakCallbackInfo<void>&)’} [-Wcast-function-type]
::   SOLINK_MODULE(target) Release/obj.target/segfault-handler.node
::   COPY Release/segfault-handler.node
:: make: Leaving directory '/build/webthings-gateway/parts/webthings-gateway/build/node_modules/segfault-handler/build'
::
:: > core-js@2.6.12 postinstall /build/webthings-gateway/parts/webthings-gateway/build/node_modules/@babel/polyfill/node_modules/core-js
:: > node -e "try{require('./postinstall')}catch(e){}"
::
:: [96mThank you for using core-js ([94m https://github.com/zloirock/core-js [96m) for polyfilling JavaScript standard library![0m
::
:: [96mThe project needs your help! Please consider supporting of core-js on Open Collective or Patreon: [0m
:: [96m>[94m https://opencollective.com/core-js [0m
:: [96m>[94m https://www.patreon.com/zloirock [0m
::
:: [96mAlso, the author of core-js ([94m https://github.com/zloirock [96m) is looking for a good job -)[0m
::
::
:: > gifsicle@5.3.0 postinstall /build/webthings-gateway/parts/webthings-gateway/build/node_modules/gifsicle
:: > node lib/install.js
::
[30/Sep/2024:13:48:52 +0000] "CONNECT raw.githubusercontent.com:443 HTTP/1.1" 200 725197 "-" "-"
:: Command failed: /build/webthings-gateway/parts/webthings-gateway/build/node_modules/gifsicle/vendor/gifsicle --version
:: /build/webthings-gateway/parts/webthings-gateway/build/node_modules/gifsicle/vendor/gifsicle: 1: ELF: not found
:: /build/webthings-gateway/parts/webthings-gateway/build/node_modules/gifsicle/vendor/gifsicle: 2: @8: not found
:: /build/webthings-gateway/parts/webthings-gateway/build/node_modules/gifsicle/vendor/gifsicle: 3: Syntax error: ")" unexpected
::
::
:: gifsicle pre-build test failed
:: compiling from source
:: Error: Command failed: /bin/sh -c autoreconf -ivf
:: autoreconf: export WARNINGS=
:: autoreconf: Entering directory '.'
:: autoreconf: configure.ac: not using Gettext
:: autoreconf: running: aclocal --force
:: Can't exec "aclocal": No such file or directory at /usr/share/autoconf/Autom4te/FileUtils.pm line 274.
:: autoreconf: error: aclocal failed with exit status: 2
::
::
::     at Promise.all.then.arr (/build/webthings-gateway/parts/webthings-gateway/build/node_modules/bin-build/node_modules/execa/index.js:231:11)
::     at process._tickCallback (internal/process/next_tick.js:68:7)
:: npm WARN rollback Rolling back gifsicle@5.3.0 failed (this is probably harmless): ENOTEMPTY: directory not empty, rmdir '/build/webthings-gateway/parts/webthings-gateway/build/node_modules/gifsicle'
:: npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.3.2 (node_modules/fsevents):
:: npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"arm64"})
::
:: npm ERR! code ELIFECYCLE
:: npm ERR! errno 1
:: npm ERR! gifsicle@5.3.0 postinstall: `node lib/install.js`
:: npm ERR! Exit status 1
:: npm ERR!
:: npm ERR! Failed at the gifsicle@5.3.0 postinstall script.
:: npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
::
:: npm ERR! A complete log of this run can be found in:
:: npm ERR!     /root/.npm/_logs/2024-09-30T13_48_57_113Z-debug.log
'override-build' in part 'webthings-gateway' failed with code 1.
Review the scriptlet and make sure it's correct.
Full execution log: '/root/.local/state/snapcraft/log/snapcraft-20240930-134533.688707.log'
Build failed
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/lpbuildd/target/build_snap.py", line 266, in run
    self.build()
  File "/usr/lib/python3/dist-packages/lpbuildd/target/build_snap.py", line 250, in build
    self.run_build_command(["snapcraft"], cwd=output_path, env=env)
  File "/usr/lib/python3/dist-packages/lpbuildd/target/operation.py", line 62, in run_build_command
    return self.backend.run(args, cwd=cwd, env=full_env, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/lpbuildd/target/lxd.py", line 718, in run
    subprocess.check_call(cmd, **kwargs)
  File "/usr/lib/python3.12/subprocess.py", line 413, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['lxc', 'exec', 'lp-jammy-arm64', '--env', 'LANG=C.UTF-8', '--env', 'SHELL=/bin/sh', '--env', 'http_proxy=http://10.10.10.1:8222/', '--env', 'https_proxy=http://10.10.10.1:8222/', '--env', 'GIT_PROXY_COMMAND=/usr/local/bin/lpbuildd-git-proxy', '--env', 'SNAPPY_STORE_NO_CDN=1', '--env', 'SNAPCRAFT_BUILD_INFO=1', '--env', 'SNAPCRAFT_IMAGE_INFO={"build-request-id": "lp-92543604", "build-request-timestamp": "2024-09-30T13:41:17Z", "build_url": "https://launchpad.net/~build.snapcraft.io/+snap/d6fc6b405a4a45eb3b743b2b7e8b0f36/+build/2613474"}', '--env', 'SNAPCRAFT_BUILD_ENVIRONMENT=host', '--env', 'SNAPCRAFT_BUILD_FOR=arm64', '--', '/bin/sh', '-c', 'cd /build/webthings-gateway && linux64 snapcraft']' returned non-zero exit status 1.
Revoking proxy token...
RUN: /usr/share/launchpad-buildd/bin/in-target scan-for-processes --backend=lxd --series=jammy --arch=arm64 SNAPBUILD-2613474
Scanning for processes to kill in build SNAPBUILD-2613474
```

--- a/snap/snapcraft.yaml
+++ b/snap/snapcraft.yaml
@@ -28,7 +28,7 @@ parts:
     source: https://github.com/WebThingsIO/gateway.git
     after: [ python-deps ]
     npm-include-node: true
-    npm-node-version: 10.24.1
+    npm-node-version: 12.22.1
     build-environment:
       - npm_config_unsafe_perm: "true"
       - NODE_ENV: "dev"

--- a/snap/snapcraft.yaml
+++ b/snap/snapcraft.yaml
@@ -34,7 +34,7 @@ parts:
       - NODE_ENV: "dev"
       - CPPFLAGS: "$CPPFLAGS -DPNG_ARM_NEON_OPT=0"
     build-packages:
-      - autoconf
+      - automake
       - build-essential
       - libbluetooth-dev
       - libboost-python-dev

--- a/snap/snapcraft.yaml
+++ b/snap/snapcraft.yaml
@@ -48,7 +48,7 @@ parts:
       craftctl default
       npm install --only-dev
       npm run-script build
-      cp -av build $CRAFT_PART_INSTALL/lib/node_modules/webthings-gateway/
+      cp -avf build $CRAFT_PART_INSTALL/lib/node_modules/webthings-gateway/
       # fix wrongly hardcoded /usr/bin/python interpreter in pagekite
       if [ -e $CRAFT_PART_INSTALL/lib/node_modules/webthings-gateway/pagekite.py ]; then
         sed -i 's;#!.*;#! /usr/bin/env python3;' \

